examples/vanilla-es6/src/store.js

```diff
@@ -0,0 +1,148 @@
+/*jshint eqeqeq:false */
+
+/**
+ * Creates a new client side storage object and will create an empty
+ * collection if no collection already exists.
+ *
+ * @param {string} name The name of our DB we want to use
+ * @param {function} callback Our fake DB uses callbacks because in
+ * real life you probably would be making AJAX calls
+ */
+export default class Store {
+	constructor(name, callback) {
+		this._dbName = name;
+
+		if (!localStorage[name]) {
+			let data = {
+				todos: []
+			};
+
+			localStorage[name] = JSON.stringify(data);
+		}
+
+		if (callback) {
+			callback.call(this, JSON.parse(localStorage[name]));
+		}
+	}
+
+	/**
+	 * Finds items based on a query given as a JS object
+	 *
+	 * @param {object} query The query to match against (i.e. {foo: 'bar'})
+	 * @param {function} callback   The callback to fire when the query has
+	 * completed running
+	 *
+	 * @example
+	 * db.find({foo: 'bar', hello: 'world'}, function (data) {
+		 *	 // data will return any items that have foo: bar and
+		 *	 // hello: world in their properties
+		 * })
+	 */
+	find(query, callback){
+		if (!callback) {
```

Comment:
Is that really a concern? It seems like all calls of `find()` and `read()` supply a callback.